validator/validators/performance/performance_validator.go

```diff
@@ -57,3 +90,153 @@ func (s *PerformanceValidator) Cleanup() error {
 
 	return nil
 }
+
+func (s *PerformanceValidator) SendPacketToDatabase(perfInfo PerformanceInformation) error {
+	var (
+		dataType               = s.vConfig.GetDataType()
+		receiver               = s.vConfig.GetPluginsConfig()
+		commitHash, commitDate = s.vConfig.GetCommitInformation()
+		agentCollectionPeriod  = fmt.Sprint(s.vConfig.GetAgentCollectionPeriod().Seconds())
+		// The secondary global index that is used for checking if there are item has already been exist in the table
+		// The performance validator will query based on the UseCaseHash to confirm if the current commit with the use case
+		// has been exist or not? If yes, merge it. If not, sending it to the database
+		// https://github.com/aws/amazon-cloudwatch-agent-test/blob/e07fe7adb1b1d75244d8984507d3f83a7237c3d3/terraform/setup/main.tf#L46-L53
+		kCheckingAttribute = []string{"CommitHash", "UseCase"}
+		vCheckingAttribute = []string{fmt.Sprint(commitHash), receiver}
+	)
+
+	err := backoff.Retry(func() error {
```

Comment:
This is to retry for case with [optimism locks](https://stackoverflow.com/questions/129329/optimistic-vs-pessimistic-locking). DynamoDB will lock the database since there are another requests going on. Therefore, we will need to retry since we always want the latest data for the performance metrics  (e.g running performance on the same commit to do an A-A testing)